testing/framework/TestCommon.py

```diff
@@ -292,14 +292,13 @@ def find(o, l):
                     return o.index(l)
                 except ValueError:
                     return None
-        contains = find(file_contents, required)
-        if not contains:
+        if find(file_contents, required) in (None,-1):
```

Comment:
when would .find() return None?

Would:
required in file_contents work as well?

Reply:
The local "find()" returns None when the substring ``required`` is not found in ``file_contents``. This is, how it was implemented previously. According to [the commend of mwichmann](https://github.com/SCons/scons/pull/3209#pullrequestreview-162071775) I decided to also handle ``-1``, as it may be returned by a custom ``find`` provided by caller for the "not found" case (e.g. ``str.find()`` behaves this way).

The ``required in file_contents`` shorthand looks good.  A workaround seems to be necessary for the ``find`` argument, which was documented to be a function returning ``None`` when the substring can't be found.

Reply:
I was thinking the local finds could be changed so they:

```
                except ValueError:
-                    return None
+                    return -1
```

The `must_contain` documentation was something I just added (commit 0cf044543) based on inspection, I assume we can certainly fix that to say it returns -1, it's an internal interface.

Reply:
From python docs:

> string.find(s, sub[, start[, end]])
> Return the lowest index in s where the substring sub is found such that sub is wholly contained in s[start:end]. Return -1 on failure. Defaults for start and end and interpretation of negative values is the same as for slices.

There's no note of returning **None**.

Reply:
Ahh.. ignore my previous comment. This is about the find function passed into various test framework methods.